File: frontend/src/objects/customer/profile/CustomerIdentity.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
import type {
  AccountStatus,
  AddressText,
  CustomerId,
  PersonName,
  PhoneNumber,
} from '@/objects/core/SharedObjects'
import type { AddressEntry } from '@/objects/customer/profile/AddressEntry'
import type { CustomerLocation } from '@/objects/customer/profile/CustomerLocation'

export type CustomerIdentity = {
  id: CustomerId
  name: PersonName
  phone: PhoneNumber
  defaultAddress: AddressText
  location?: CustomerLocation
  addresses: AddressEntry[]
  accountStatus: AccountStatus
}

File: backend/src/main/scala/services/customer/objects/CustomerIdentity.scala
package services.customer.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given

import system.objects.*
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.*

final case class CustomerIdentity(
    id: CustomerId,
    name: PersonName,
    phone: PhoneNumber,
    defaultAddress: AddressText,
    location: Option[CustomerLocation],
    addresses: List[AddressEntry],
    accountStatus: AccountStatus,
)
object CustomerIdentity:
  given Encoder[CustomerIdentity] = deriveEncoder
  given Decoder[CustomerIdentity] = deriveDecoder
